git/squash.md

```diff
@@ -0,0 +1,16 @@
+# How and Why to Squash Merge your Pull Requests
+
+[![How and Why to Squash Merge your Pull Requests](http://i3.ytimg.com/vi/41EITMtWUZM/maxresdefault.jpg)](https://www.youtube.com/watch?v=41EITMtWUZM "How and Why to Squash Merge your Pull Requests")
+
+When merging a feature branch PR into master, we recommend you squash your commits into one (or just a few) commits. We do this to avoid having a messy history full of "fix typo" and "lint fix" and "oops" commits. As a rule of thumb, most PRs should squash down to a single commit with a well-written commit message explaining why a change is happening.
+
+On GitHub, you can use the "Squash Merge" option when merging a commit to automatically squash all commits into one, with the option to rewrite the commit message.
```

Comment:
Would an image of the "Squash Merge" button be helpful here?

Reply:
Good idea!